iroha/src/smartcontracts/isi/permissions.rs

```diff
@@ -12,61 +12,78 @@ use super::prelude::WorldTrait;
 use super::Evaluate;
 use crate::prelude::*;
 
+/// Operation for which the permission should be checked.
+pub trait ShouldCheckPermission {}
+
+impl ShouldCheckPermission for Instruction {}
+
+impl ShouldCheckPermission for QueryBox {}
+
```

Comment:
May be we should move it to smartcontracts module? Or it will be specific to isi?

Reply:
Also `Permitable` might be a good name

Reply:
It will be specific to isi, wasm will call our queries or instructions and therefore use permissions through them